### ISSUE TITLE
Support for complex numbers

### ISSUE THREAD

**Issue by karlrupp** [viennacl/viennacl-dev#15]
Quite a number of applications relies on complex arithmetics. The difficulty is the lack of native support for complex_t in OpenCL, thus all operations need to be emulated. Addition and subtraction are easy, but multiplication and divisions are tricky. Emulation of sqrt() and the like is also required.

**Comment by Kerkil** [comment]
Has there been progress on this front? (I realize that this is an old thread.)

**Comment by karlrupp** [comment]
No, there hasn't been significant progress recently, even though it scheduled for future releases:
https://github.com/viennacl/viennacl-dev/wiki/ViennaCL-Roadmap

**Comment by romanlarionov** [comment]
Does anybody know if there are good alternatives to ViennaCL that supports double complex types, similar to how cuBLAS does? This library seems perfect for my problem, but no support for double complex types is a deal breaker for me.
